azure-spring-boot/src/main/java/com/microsoft/azure/telemetry/TelemetrySender.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.telemetry;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.spring.support.GetHashMac;
+import com.microsoft.azure.utils.PropertyLoader;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.*;
+import org.springframework.util.Assert;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Map;
+
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON;
+
+@Slf4j
+public class TelemetrySender {
+
+    private static final String TELEMETRY_TARGET_URL = "https://dc.services.visualstudio.com/v2/track";
+
+    private static final String PROJECT_INFO = "spring-boot-starter/" + PropertyLoader.getProjectVersion();
+
+    private static final int RETRY_LIMIT = 3; // Align the retry times with sdk
+
+    private ResponseEntity<String> executeRequest(final TelemetryEventData eventData) {
+        final HttpHeaders headers = new HttpHeaders();
+        final ObjectMapper mapper = new ObjectMapper();
+
+        headers.add(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON.toString());
+
+        try {
+            final RestTemplate restTemplate = new RestTemplate();
```

Comment:
Do we need to create a `RestTemplate` every time? `RestTemplate` is thread safe, right?

Reply:
make sense, will move to class level.